src/components/Thanks/SingleVersion/GoalEntrypoint.vue

```diff
@@ -59,45 +64,174 @@
 import HandsPlant from '#src/assets/images/thanks-page/hands-plant.svg';
 import ThumbUp from '#src/assets/images/thanks-page/thumbs-up.svg';
 import GoalSelector from '#src/components/Thanks/SingleVersion/GoalSelector';
+import GoalSettingModal from '#src/components/MyKiva/GoalSettingModal';
+import useGoalData from '#src/composables/useGoalData';
+
+const apollo = inject('apollo');
+const $kvTrackEvent = inject('$kvTrackEvent');
+const router = useRouter();
+
+const {
+	storeGoalPreferences,
+} = useGoalData({ apollo });
 
 const props = defineProps({
+	/**
+	 * Whether the component is in a loading state
+	 */
 	loading: {
 		type: Boolean,
 		default: false,
 	},
-	isThanksPage: {
-		type: Boolean,
-		default: true,
+	/**
+	 * total number of loans made by the user
+	 */
+	totalLoans: {
+		type: Number,
+		default: 0,
+	},
+	/**
+	 * Tiered achievements data
+	 */
+	tieredAchievements: {
+		type: Array,
+		default: () => ([]),
 	},
 });
 
+const isThanksPage = ref(false);
+const showGoalModal = ref(false);
+
+const goalOptions = ref([
+	{ loansNumber: 3, optionText: 'Start strong', selected: false },
+	{
+		loansNumber: 4, highlightedText: 'Recommended', optionText: 'Extra mile', selected: true,
+	},
+	{ loansNumber: 5, optionText: 'Trailblazing!', selected: false },
+]);
+
 const titleText = computed(() => {
-	return props.isThanksPage
+	return isThanksPage.value
 		? 'Thank you!'
-		: 'Last year, you helped <span class="tw-text-eco-green-3">2 women</span> shape their futures!';
+		: 'Lenders like you help <br><span class="tw-text-eco-green-3">3 women</span> a year';
 });
 
 const subtitleText = computed(() => {
-	return props.isThanksPage
+	return isThanksPage.value
 		? 'Your 2026 commitment means more lives transformed!'
 		: 'How many loans will you make this year?';
 });
 
-const buttonText = computed(() => (props.isThanksPage ? 'Track my progress' : 'Set 2026 goal'));
+const buttonText = computed(() => (isThanksPage.value ? 'Track my progress' : 'Set 2026 goal'));
+
+const categoriesLoanCount = computed(() => {
+	const { getAllCategoryLoanCounts } = useBadgeData();
+	return getAllCategoryLoanCounts(props.tieredAchievements);
+});
+
+const selectedTarget = computed(() => {
+	const selectedOption = goalOptions.value.find(option => option.selected);
+	return selectedOption.loansNumber;
+});
+
+const updateOptionSelection = selectedIndex => {
+	goalOptions.value = goalOptions.value.map((option, index) => ({
+		...option,
+		selected: index === selectedIndex,
+	}));
+
+	const trackingProperties = ['same-as-last-year', 'a-little-more', 'double'];
+	$kvTrackEvent(
+		'post-checkout',
+		'click',
+		'set-goal-amount',
+		trackingProperties[selectedIndex]
+	);
+};
+
+const editGoal = () => {
+	showGoalModal.value = true;
+	$kvTrackEvent(
+		'post-checkout',
+		'click',
+		'edit-goal-category'
+	);
+};
+
+const setGoal = async preferences => {
+	await storeGoalPreferences(preferences);
+	isThanksPage.value = true;
```

Comment:
This flag is currently set when the goal is set, but won't we need this on page load? Also the entry points are similar but different, it's going to be interesting re-using this component.

Reply:
Initially thought the same, but based on flow described in figma file, is just a thank you message when you have set a goal on the entrypoint. So, we just need to set this flag when the goal is set there.

https://www.figma.com/design/ARJWLwdnFJ84FNSvPX4wYz/Impact-Goals-V2-Oct.2025--MK-?node-id=738-10769&m=dev

Reply:
I wonder if the flag should be changed then to something like `isGoalSetOnThanksPage`.